packages/twenty-server/src/engine/core-modules/analytics/services/clickhouse.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+
+import { ClickHouseClient, createClient } from '@clickhouse/client';
+import { Pageview, Event } from 'twenty-analytics';
+
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+@Injectable()
+export class ClickhouseService {
+  private clickhouseClient: ClickHouseClient;
+
+  constructor(private readonly environmentService: EnvironmentService) {
+    this.clickhouseClient = createClient({
+      url: environmentService.get('CLICKHOUSE_URL'),
+      database: environmentService.get('CLICKHOUSE_DB'),
+    });
+  }
+
+  async insert(data: Pageview | Event) {
+    try {
+      await this.clickhouseClient.insert({
```

Comment:
Is this the asyncInsert we talked about?

Reply:
Know that the async insert and the bulk insert are implemented